browsermob-rest/src/main/java/net/lightbody/bmp/proxy/bricks/ProxyResource.java

```diff
@@ -407,7 +407,17 @@ public Reply<?> addResponseFilter(@Named("port") int port, Request<String> reque
         String script = getEntityBodyFromRequest(request);
         requestResponseFilter.setResponseFilterScript(script);
 
-        proxy.addResponseFilter(requestResponseFilter);
+        String bufferSize = request.param("bufferSize");
+        if (bufferSize != null && !bufferSize.isEmpty()) {
```

Comment:
Return early instead of conditional checks, and writing repeated code.

Reply:
Can't return early, because we need to perform some actions depending on this condition.